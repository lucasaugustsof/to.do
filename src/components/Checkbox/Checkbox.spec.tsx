import { describe, it, vi, expect } from 'vitest'

import { render, screen } from '@testing-library/react'
import { userEvent } from '@testing-library/user-event'

import { Checkbox } from '.'

const onClick = vi.fn()

describe('Components/Checkbox', () => {
  it('should render the Checkbox component correctly', () => {
    const { asFragment } = render(<Checkbox onClick={onClick} checked />)

    expect(asFragment()).toMatchSnapshot()
  })

  it('should call the click event at least 1 time', async () => {
    render(<Checkbox onClick={onClick} checked={false} />)

    const checkboxComponent = screen.getByRole('button')
    await userEvent.click(checkboxComponent)

    expect(onClick).toHaveBeenCalledOnce()
  })

  it('should render a check icon when the checked property is true', () => {
    render(<Checkbox onClick={onClick} checked />)

    const checkboxComponent = screen.getByRole('button').children[0]
    const checkboxCheckIcon = checkboxComponent.children[0]

    expect(checkboxComponent.getAttribute('data-checked')).toBeTruthy()
    expect(checkboxCheckIcon.getElementsByTagName('svg')).toBeTruthy()
  })
})
