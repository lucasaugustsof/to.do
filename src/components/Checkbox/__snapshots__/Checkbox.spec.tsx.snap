// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Components/Checkbox > should render the Checkbox component correctly 1`] = `
<DocumentFragment>
  <button
    class="_container_9097fe"
  >
    <div
      data-checked="true"
    >
      <svg
        fill="currentColor"
        height="10"
        viewBox="0 0 256 256"
        width="10"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M229.66,77.66l-128,128a8,8,0,0,1-11.32,0l-56-56a8,8,0,0,1,11.32-11.32L96,188.69,218.34,66.34a8,8,0,0,1,11.32,11.32Z"
        />
      </svg>
    </div>
  </button>
</DocumentFragment>
`;
